### PR TITLE
[bitnami/sealed-secrets] add missing features from the upstream sealed secrets chart

### PR DESCRIPTION
**Pull request by alemorcuq**
### Description of the change
Add missing features from the upstream chart

### Benefits
This chart has the same features as upstream

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/sealed-secrets/templates/deployment.yaml [hunk 1]
```diff
@@ -85,7 +89,54 @@ spec:
...
+            - {{ printf "%s" (include "sealed-secrets.secretName" .) | quote }}
+            {{- if .Values.updateStatus }}
+            - "--update-status"
+            {{- end }}
+            {{- if .Values.skipRecreate }}
+            - "--skip-recreate"
+            {{- end }}
+            {{- if .Values.keyrenewperiod }}
+            - "--key-renew-period"
+            - {{ .Values.keyrenewperiod | quote }}
```
Comment (FraPazGal) [inline bitnami/sealed-secrets/templates/deployment.yaml:101]:
```suggestion
            {{- if .Values.keyRenewPeriod }}
            - "--key-renew-period"
            - {{ .Values.keyRenewPeriod | quote }}
```

File: bitnami/sealed-secrets/templates/deployment.yaml [hunk 2]
```diff
@@ -85,7 +89,54 @@ spec:
...
+            - "--rate-limit-burst"
+            - {{ .Values.rateLimitBurst | quote }}
+            {{- end }}
+            {{- if .Values.additionalNamespaces }}
+            - "--additional-namespaces"
+            - {{ join "," .Values.additionalNamespaces | quote }}
+            {{- end }}
+            {{- if $.Values.privateKeyAnnotations }}
+            {{- $privatekeyAnnotations := ""}}
+            {{- range $k, $v := $.Values.privateKeyAnnotations }}
```
Comment (FraPazGal) [inline bitnami/sealed-secrets/templates/deployment.yaml:117]: Are we using here the right context? Both `.Values` calls are outside the `range`, unless I'm mistaken we should use the following:

```suggestion
            {{- if .Values.privateKeyAnnotations }}
            {{- $privatekeyAnnotations := ""}}
            {{- range $k, $v := .Values.privateKeyAnnotations }}
```

Same comment applies to the logic used for `privateKeyLabels` below.

File: bitnami/sealed-secrets/values.yaml [hunk 3]
```diff
@@ -477,6 +516,18 @@ rbac:
...
   pspEnabled: false
+  ## @param rbac.clusterRole Specifies whether the Cluster Role resource should be created
+  ##
+  clusterRole: true
+  ## @param rbac.clusterRoleName Specifies the name for the Cluster Role resource
+  ##
+  clusterRoleName: ""
+  ## @param rbac.namespacedRoles Specifies whether the namespaced Roles should be created (in each of the specified additionalNamespaces)
+  ##
+  namespacedRoles: false
```
Comment (FraPazGal) [inline bitnami/sealed-secrets/values.yaml:527]: Just a recommendation, should we detail here that using `clusterRole=true` and `namespacedRoles=true` will result in no clusterrole nor roles being created? I understand the logic but maybe users can be confused if they try using both.

File: bitnami/sealed-secrets/templates/role.yaml [hunk 4]
```diff
@@ -57,3 +64,61 @@ rules:
...
+kind: Role
+metadata:
+  name: {{ printf "%s" (include "sealed-secrets.namespacedRoleName" .) }}
+  namespace: {{ $additionalNamespace }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+  {{- if .Values.rbac.labels }}
+  {{- include "common.tplvalues.render" ( dict "value" .Values.rbac.labels "context" $) | nindent 4 }}
+  {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
```
Comment (FraPazGal) [inline bitnami/sealed-secrets/templates/role.yaml:80]: These are inside the range's context so we'll need to modify them as so:

```suggestion
  labels: {{- include "common.labels.standard" ( dict "customLabels" $.Values.commonLabels "context" $ ) | nindent 4 }}
  {{- if $.Values.rbac.labels }}
  {{- include "common.tplvalues.render" ( dict "value" $.Values.rbac.labels "context" $) | nindent 4 }}
  {{- end }}
  {{- if $.Values.commonAnnotations }}
  annotations: {{- include "common.tplvalues.render" ( dict "value" $.Values.commonAnnotations "context" $ ) | nindent 4 }}
```

Same suggestion for the `clusterrole.yaml` template.

File: bitnami/sealed-secrets/templates/role.yaml [hunk 5]
```diff
@@ -57,3 +64,61 @@ rules:
   {{- include "common.tplvalues.render" ( dict "value" .Values.rbac.serviceProxier.rules "context" $ ) | nindent 2 }}
   {{- end }}
 {{- end }}
+{{ if and (and .Values.rbac.create .Values.rbac.namespacedRoles) (not $.Values.rbac.clusterRole) }}
```
Comment (FraPazGal) [inline bitnami/sealed-secrets/templates/role.yaml:67]:
```suggestion
{{ if and (and .Values.rbac.create .Values.rbac.namespacedRoles) (not .Values.rbac.clusterRole) }}
```
Same suggestion for the `clusterrole.yaml` template.

**Review by FraPazGal — COMMENTED** [review]
The new features look good @alemorcuq! I have added a few comments as a review, please check them whenever you can

File: bitnami/sealed-secrets/values.yaml [hunk 6]
```diff
@@ -91,6 +91,45 @@ commandArgs: []
...
+updateStatus: true
+## @param skipRecreate Specifies whether the Sealed Secrets controller should skip recreating removed secrets
+## Setting it to true allows to optionally restore backward compatibility in low priviledge
+## environments when old versions of the controller did not require watch permissions on secrets
+## for secret re-creation.
+##
+skipRecreate: false
+## @param keyRenewPeriod Specifies key renewal period. Default 30 days. e.g keyRenewPeriod: "720h30m"
+##
+keyRenewPeriod: ""
```
[the remark this answers is not in the document]
Comment (alemorcuq) [reply bitnami/sealed-secrets/values.yaml:114]: Yes, it's a builtin default. Should I leave it in the comment or remove it?

[the remark this answers is not in the document]
Comment (FraPazGal) [reply bitnami/sealed-secrets/values.yaml:114]: I think we can leave it in the comment; no strong thoughts on my side

**Review by FraPazGal — APPROVED** [review]
LGTM! Thanks for the changes @alemorcuq

**Pull request merged**